content/blog/2020-01-ggplot2-3-3-0/index.Rmarkdown

```diff
@@ -0,0 +1,231 @@
+  comment = "#>"
+)
+```
+
+We're so happy to announce the release of ggplot2 3.3.0 on CRAN. ggplot2 is a system for declaratively creating graphics, based on *The Grammar of Graphics*. You provide the data, tell ggplot2 how to map variables to aesthetics, what graphical primitives to use, and it takes care of the details.
+
+This release is a substantial release with both big new features and internal rewrites. It contains most of the work Dewey Dunnington did as part of his internship at RStudio, along with contributions from many others. Dewey will continue working on ggplot2 as part of the core team, and we are very lucky to have him.
+
+## New features
+The last ggplot2 release was mainly a performance-oriented release. This one, on the other hand is packed with features, big and small. There is also a slew of bug-fixes, but as that make for less interesting reading, please consult the NEWS.md file for those. Further, we have now removed reshape2 from the dependencies, which further removes the indirect dependencies on plyr, stringi and stringr. We have been shaving off dependencies in ggplot2 over a range of releases, and with the removal of reshape2 we are now close to being as lean as we believe is possible.
+
+### Rewrite of axis code
+While Dewey has worked on a lot of different parts of the ggplot2 code base, the lion share has been concerned with a rewrite of the positional guide (axis) internals. While, at a high level, axes and legends are equivalent (they are both guides used for scales), this has not been true for the underlying code. With this release we have streamlined the implementation considerably, and paved the way for a full guide rewrite in a future release (guides are one of the last part waiting to be updated to ggproto). Apart from making our life as ggplot2 developers easier, the rewrite also comes with a slew of user facing improvements and features. All of this is contained in the new `guide_axis()` function that works equivalently to e.g. `guide_legend()`.
+
+```{r, message=FALSE}
+library(ggplot2)
+
+p <- ggplot(mpg) +
+  geom_bar(aes(x = manufacturer)) + 
+  theme(axis.text.x = element_text(size = 11))
+
+# Overlapping labels
+plot(p)
+```
+
+```{r}
+# Use guide_axis to dodge the labels
+p + 
+  scale_x_discrete(guide = guide_axis(n.dodge = 2))
+```
+
+```{r}
+# Or to remove overlapping labels
+p + 
+  scale_x_discrete(guide = guide_axis(check.overlap = TRUE))
+```
+
+### New bin scale
+While ggplot2 has a lot of different scales, it has only ever had two different scale types: *continuous* and *discrete*. This all changes with this release, as a new binning scale type has been added. A binning scale takes continuous data and makes it discrete by assigning each data point to a bin. It thus exist somewhere between the two existing scale types and e.g. allow continuous data to be used with discrete palettes. Being a fundamental scale type it is available to all aesthetics, both positional and otherwise:
```

Comment:
```suggestion
While ggplot2 has a lot of different scales, it has only ever had two different scale types: *continuous* and *discrete*. This all changes with this release, as a new binning scale type has been added. A binning scale takes continuous data and makes it discrete by assigning each data point to a bin. It thus exists somewhere between the two existing scale types and e.g. allows continuous data to be used with discrete palettes. Being a fundamental scale type, it is available to all aesthetics, both positional and otherwise:
```

Reply:
Added commas (_damn you GH diff!_)